Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentMode.cpp

```diff
@@ -121,6 +132,741 @@ namespace PhysX
         m_subModes[m_subMode]->Teardown(GetEntityComponentIdPair());
     }
 
+    void JointsComponentMode::Reflect(AZ::ReflectContext* context)
+    {
+        AzToolsFramework::ComponentModeFramework::ReflectEditorBaseComponentModeDescendant<JointsComponentMode>(context);
+    }
+
+    void JointsComponentMode::RegisterActions()
+    {
+        auto actionManagerInterface = AZ::Interface<AzToolsFramework::ActionManagerInterface>::Get();
+        AZ_Assert(actionManagerInterface, "JointsComponentMode - could not get ActionManagerInterface on RegisterActions.");
+
+        auto hotKeyManagerInterface = AZ::Interface<AzToolsFramework::HotKeyManagerInterface>::Get();
+        AZ_Assert(hotKeyManagerInterface, "JointsComponentMode - could not get HotKeyManagerInterface on RegisterActions.");
+
+        // Switch to Translation Sub-Mode
+        {
+            constexpr AZStd::string_view actionIdentifier = "o3de.action.jointsComponentMode.switchToTranslationSubMode";
+            AzToolsFramework::ActionProperties actionProperties;
+            actionProperties.m_name = SubModeData::TranslationTitle;
+            actionProperties.m_description = SubModeData::TranslationToolTip;
+            actionProperties.m_category = "Joints Component Mode";
+
+            actionManagerInterface->RegisterAction(
+                EditorMainWindowActionContextIdentifier,
+                actionIdentifier,
+                actionProperties,
+                []
+                {
+                    auto componentModeCollectionInterface = AZ::Interface<AzToolsFramework::ComponentModeCollectionInterface>::Get();
+                    AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");
+
+                    componentModeCollectionInterface->EnumerateActiveComponents(
+                        [](const AZ::EntityComponentIdPair&, const AZ::Uuid&)
+                        {
+                            auto componentModeCollectionInterface =
+                                AZ::Interface<AzToolsFramework::ComponentModeCollectionInterface>::Get();
+                            AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");
+
+                            componentModeCollectionInterface->EnumerateActiveComponents(
+                                [](const AZ::EntityComponentIdPair& entityComponentIdPair, const AZ::Uuid&)
+                                {
+                                    JointsComponentModeRequestBus::Event(
+                                        entityComponentIdPair, &JointsComponentModeRequests::SetCurrentSubMode,
+                                        JointsComponentModeCommon::SubComponentModes::ModeType::Translation);
+                                }
+                            );
+                        }
+                    );
+                }
+            );
```

Comment:
I might be reading it wrong since it's a bit tricky to read 3 embedded lambdas, but it seems like the second and third lambda are doing the same (both of them calling `componentModeCollectionInterface->EnumerateActiveComponents`)? Can the last lambda be removed?

````
            actionManagerInterface->RegisterAction(
                EditorMainWindowActionContextIdentifier,
                actionIdentifier,
                actionProperties,
                []
                {
                    auto componentModeCollectionInterface = AZ::Interface<AzToolsFramework::ComponentModeCollectionInterface>::Get();
                    AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");

                    componentModeCollectionInterface->EnumerateActiveComponents(
                        [](const AZ::EntityComponentIdPair& entityComponentIdPair, const AZ::Uuid&)
                        {
                            JointsComponentModeRequestBus::Event(
                                entityComponentIdPair, &JointsComponentModeRequests::SetCurrentSubMode,
                                JointsComponentModeCommon::SubComponentModes::ModeType::Translation);
                        }
                    );
                }
            );
````

Same in other places in the code.

Reply:
You're right, the nested Enumerates were a copy-paste mistake... I thought I fixed it but something may have gone wrong. Fixing it again!

Reply:
+1 to @moraaar's comment